packages/core/test/request-builder/header-builder/auth-headers.spec.ts

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { MapType } from '@sap-cloud-sdk/util';
+import { buildHeadersForDestination, Destination } from '../../../src';
+import { muteLoggers } from '../../test-util/mute-logger';
+import { buildAuthorizationHeader } from '../../../src/request-builder/header-builder/auth-headers';
+
+describe('Authorization header builder', () => {
+  beforeAll(() => {
+    muteLoggers('auth-headers');
+  });
+
+  it('does not throw on NoAuthentication', async () => {
+    await expect(
+      buildAuthorizationHeader({
+        url: 'https://example.com',
+        authentication: 'NoAuthentication'
+      })
+    ).resolves.not.toThrow();
+  });
+
+  it('does not throw on ClientCertificateAuthentication', async () => {
+    await expect(
+      buildAuthorizationHeader({
+        url: 'https://example.com',
+        authentication: 'ClientCertificateAuthentication'
+      })
+    ).resolves.not.toThrow();
+  });
+
+  it('defaults to NoAuthentication', async () => {
+    await expect(
```

Comment:
shouldn't there be an assertion that there is no auth header?

Reply:
100% agree. I think we have a lot of potential to improve our test strategy. In the end this tests what we want implicitly, but is way more complicated than it should be in my opinion. I just didn't want to touch this more than I already did.